scripts/obtain/install.sh

```diff
@@ -4,56 +4,25 @@
 
 # Note: This script should be compatible with the dash shell used in Ubuntu. So avoid bashisms! See https://wiki.ubuntu.com/DashAsBinSh for more info
 
-# This is a herestring Everything from the line AFTER the "read" until the line containing ONLY "EOF" is part of the string
-# The quotes around "EOF" ensure that $ is not interpreted as a variable expansion. The indentation of the herestring must be
-# kept exactly consistent
-LINK_SCRIPT_CONTENT=$(cat <<-"EOF"
-#!/usr/bin/env bash
-
-SOURCE="${BASH_SOURCE[0]}"
-while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
-  DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
-  SOURCE="$(readlink "$SOURCE")"
-  [[ "$SOURCE" != /* ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
-done
-DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
-
-if [ -z "$DOTNET_TOOLS" ]; then
-    # We should be in $PREFIX/bin, so just get the directory above us.
-    PREFIX="$( cd -P "$DIR/.." && pwd)"
-
-    # The tools are in $PREFIX/share/dotnet/cli by default
-    DOTNET_TOOLS_DEFAULT=$PREFIX/share/dotnet/cli
-
-    # Check the default location
-    if [ -d "$DOTNET_TOOLS_DEFAULT" ]; then
-        export DOTNET_TOOLS=$DOTNET_TOOLS_DEFAULT
-    else
-        echo "error: the .NET tools installation appears to be corrupt!" 1>&2
-        echo "error: specifically, the tools could not be located in the $DOTNET_TOOLS_DEFAULT directory" 1>&2
-        exit 1
-    fi
-fi
-
-MY_NAME=$(basename ${BASH_SOURCE[0]})
-MY_TARGET=$DOTNET_TOOLS/bin/$MY_NAME
-
-if [ ! -e "$MY_TARGET" ]; then
-    echo "error: the tool $MY_TARGET cannot be found" 1>&2
-    exit 1
-fi
-
-if [ ! -x "$MY_TARGET" ]; then
-    echo "error: the tool $MY_TARGET is not executable" 1>&2
-    exit 1
-fi
-
-exec "$MY_TARGET" "$@"
-EOF
-)
-
-#setup some colors to use. These need to work in fairly limited shells, like the Ubuntu Docker container where there are only 8 colors.
-#See if stdout is a terminal
+# Stop script on NZEC
+set -e
+# Stop script if unbound variable found (use ${var:-} if intentional)
+set -u
+# By default cmd1 | cmd2 returns exit code of cmd2 regardless of cmd1 success
+# This is causing it to fail
+set -o pipefail
+
+# Use in the the functions: eval $invocation
+invocation='say_verbose "Calling: ${red:-}${FUNCNAME[0]} ${green:-}$*${normal:-}"'
```

Comment:
I would reserve red usually fir error messages. How about yellow since this just informational?